pkg/apis/resourcedefinition/helper.go

```diff
@@ -0,0 +1,506 @@
+package resourcedefinition
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"reflect"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/tidwall/gjson"
+	"golang.org/x/exp/maps"
+	"golang.org/x/exp/slices"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/seal-io/walrus/pkg/dao/model"
+	"github.com/seal-io/walrus/pkg/dao/model/templateversion"
+	"github.com/seal-io/walrus/pkg/dao/types"
+	"github.com/seal-io/walrus/pkg/dao/types/object"
+	"github.com/seal-io/walrus/pkg/templates/openapi"
+	"github.com/seal-io/walrus/utils/json"
+	"github.com/seal-io/walrus/utils/pointer"
+)
+
+const (
+	variablesSchemaKey = "variables"
+	outputsSchemaKey   = "outputs"
+
+	originalExtensionKey = "x-walrus-original"
```

Comment:
Can we use the existed extension key?
https://github.com/seal-io/walrus/blob/6200ad68e4725668a3ab4c9f355cd95fe4b37f51/pkg/templates/openapi/ext.go#L89